app/services/pupil_premium_importer.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+require "csv"
+
+class PupilPremiumImporter
+  def initialize(logger, start_year = Time.zone.now.year, source_file = "")
+    @logger = logger
+    @start_year = start_year
+    @source_file = source_file
+  end
+
+  def run
+    CSV.foreach(data_file, headers: true, encoding: "ISO-8859-1:UTF-8") do |row|
+      update_school_premium(row)
+    end
+  end
+
+private
+
+  def data_file
+    @source_file || __dir__ + "/../../data/pupil_premium.csv"
```

Comment:
This will probably fail on native windows. Try and use something like this Rails.root.join('tmp', 'blah', 'caching-dev.txt' ) https://apidock.com/rails/Rails/root/class